src/components/Dialog/Transitions.tsx

```diff
@@ -0,0 +1,11 @@
+import React from "react"
+import { TransitionProps } from "@material-ui/core/transitions/transition"
+import Slide from "@material-ui/core/Slide"
+
+export const SlideUpTransition = React.forwardRef((props: TransitionProps, ref) => (
```

Comment:
Interesting idea - one more indirection, use case -> theme -> concrete transition -> abstract transition.

we need to be careful to not overdo the abstractioning, but this makes sense 👍